dm/dm-manage-schema.md

```diff
@@ -7,30 +7,38 @@ summary: Learn how to manage the schema of the table to be migrated in DM.
 
 This document describes how to manage the schema of the table in DM during migration using [dmctl](/dm/dmctl-introduction.md).
 
+When DM performs incremental replication, it first reads the upstream binlog, then creates SQL statements and executes them in the downstream. However, the upstream binlog does not contain the complete table schema. To generate the SQL statements, DM maintains internally the schema information of the table to be migrated. This is called the internal table schema.
+
+To deal with some special occasions, or to handle migration interruptions caused by mismatch of the table schemas, DM provides the `binlog-schema` command to obtain, modify, and delete the internal table schema.
+
 ## Implementation principles
 
-When you migrate tables using DM, DM performs the following operations on the table schema:
+The internal table schema comes from the following sources:
 
-- For full export and import, DM directly exports the upstream table schema of the current time to SQL files and applies the table schema to the downstream.
+1. For full data migration (`task-mode=all`), the migration task goes through three stages: dump/load/sync, which means full export, full import, and incremental replication. In the dump stage, DM exports the table schema information along with the data and automatically creates the corresponding table in the downstream. In the sync stage, this table schema is used as the starting table scheme for incremental replication.
+2. In the sync stage, when DM handles DDL statements such as `ALTER TABLE`, it updates the internal table schema at the same time.
+3. If the task is an incremental migration (`task-mode=incremental`), in which the downstream has completed creating the table to be migrated, DM obtains the table schema information from the downstream database. This behavior varies with DM versions.
 
-- For incremental replication, the whole data link contains the following table schemas, which might be the same or different:
+For incremental replication, the schema maintenance is complicated. The whole data link contains the following table schemas, which might be the same or different:
```

Comment:
This sentence is a little confusing:
`The whole data link contains the following table schemas, which might be the same or different`. 
I think it is intended to say that the data link may include the following table schemas or include other table schemas.
```suggestion
For incremental replication, schema maintenance is complicated. The whole data link contains the following table schemas, which might be the same or different:
```

Reply:
You're right about the sentence being confusing...

However, it doesn't look like there are any other table schemas involved. 

What the author tries to say is that **the four table schemas in the diagram** (schema-U, schema-B, schema-I, schema-D) **are usually consistent with one another** (see L31), but **sometimes might be inconsistent** (see L35).

I try to rephrase this sentence, PTAL:

```suggestion
For incremental replication, schema maintenance is complicated. During the whole data replication, the following four table schemas are involved. These schemas might be the consistent or inconsistent with one another:
```

Reply:
LGTM

Reply:
> You're right about the sentence being confusing...
> 
> However, it doesn't look like there are any other table schemas involved.
> 
> What the author tries to say is that **the four table schemas in the diagram** (schema-U, schema-B, schema-I, schema-D) **are usually consistent with one another** (see L31), but **sometimes might be inconsistent** (see L35).
> 
> I try to rephrase this sentence, PTAL:

LGTM

Reply:
LGTM